packages/react-native-gesture-handler/src/web/handlers/GestureHandler.ts

```diff
@@ -541,12 +582,22 @@ export default abstract class GestureHandler implements IGestureHandler {
       timeStamp: Date.now(),
     };
 
-    const { onGestureHandlerEvent }: PropsRef = this.propsRef
-      .current as PropsRef;
+    const { onGestureHandlerEvent }: PropsRef = this.propsRef!.current;
 
     invokeNullableMethod(onGestureHandlerEvent, cancelEvent);
   }
 
+  protected ensurePropsRef(): this is this & {
+    propsRef: React.RefObject<PropsRef>;
+  } {
```

Comment:
Wouldn't it work if we extract it to separate type? Like `GestureHandlerWithPropsRef` or something similar?